feat: Handle first admin invite to org

SuaYoo opened the pull request:
WIP frontend for https://github.com/webrecorder/browsertrix/issues/1874

<!-- Fixes #issue_number -->

### Changes

- Limits superadmin to choosing admin role when inviting the first user to an org
- Adds org rename step after account registration
- For existing users, accepts invitation with optional org name and slug

SuaYoo committed:
update role select

SuaYoo committed:
handle first org admin

SuaYoo committed:
disable new org creation

SuaYoo committed:
add additional fields

SuaYoo committed:
create separate org setup step

SuaYoo committed:
update accept invite

SuaYoo committed:
pass org name and slug to accept invite

SuaYoo committed:
convert invite to tailwind component

SuaYoo committed:
rename org

SuaYoo committed:
pass org id

SuaYoo committed:
handle org rename

SuaYoo committed:
handle invalid invite

SuaYoo committed:
handle accept invite

SuaYoo committed:
handle superadmin invite

SuaYoo committed:
always show org name for first admin

SuaYoo committed:
redirect to new org

SuaYoo committed:
show steps

ikreymer committed:
lint fix

ikreymer commented:
The 2nd step should handle the `duplicate_org_name` error slightly better, the button gets stuck on the spinner:
<img width="749" alt="Screenshot 2024-06-24 at 5 16 19 PM" src="https://github.com/webrecorder/browsertrix/assets/1015759/a4b83b5d-c507-437f-805c-1e8f91013681">

ikreymer committed:
undo optional rename in accept invite api
instead, set the org default name to '<USERNAME>'s archive' if name == id and first admin is added

ikreymer commented:
As discussed, now having the backend auto-set the org name if the org name == org id and first admin is being added.

However, it looks like frontend is still use the original org name / slug on the second screen, instead of the updated ones returned from the `/register` endpoint, which updates them. See example:

<img width="1434" alt="Screenshot 2024-06-24 at 7 03 01 PM" src="https://github.com/webrecorder/browsertrix/assets/1015759/b6f5d82a-d515-4dfb-bc37-f5d5521fcbc0">

To repro:
- Create a new org / use existing org and change Org Name to be the Org Id
- Remove all users
- Invite new user

ikreymer commented:
The 'user already exists' path seems to be broken at the moment, never showing the 'Accept Invitation' dialog, and going directly to:

<img width="1128" alt="Screenshot 2024-06-24 at 7 10 30 PM" src="https://github.com/webrecorder/browsertrix/assets/1015759/a6e7726c-3861-490b-a870-6eded63bf468">

Overall, the flow should be symmetrical:

```
If user already exists:
   If firstOrgAdmin set:
        Show two part screen, first invitation accept, then org rename (with updated defaults set after invitation is accepted)
   else:
        Show one part screen, invitation accept
 else:
    if firstOrgAdmin set:
        Show two part screen, first registration screen, then org rename (with updated defaults set after registration is complete)
    else:
        Show one part screen, user registration
 ```

SuaYoo committed:
update copy

SuaYoo committed:
use name returned from sign up

SuaYoo committed:
make accept two steps

SuaYoo committed:
add org description

SuaYoo committed:
adjust spacing

SuaYoo committed:
handle duplicate org name

SuaYoo committed:
Merge branch 'issue-team-7-invite-org-details' into issue-team-7-invite-org-details--frontend
